test/cpp/security/tls_certificate_verifier_test.cc

```diff
@@ -161,28 +161,15 @@ TEST(TlsCertificateVerifierTest,
   EXPECT_EQ(sync_status.error_message(), "Hostname Verification Check failed.");
 }
 
-TEST(TlsCertificateVerifierTest,
-     NoOpCertificateVerifierSucceedsWithoutVerifiedRootCertSubject) {
-  grpc_tls_custom_verification_check_request request;
-  memset(&request, 0, sizeof(request));
-  auto verifier = std::make_shared<NoOpCertificateVerifier>();
-  TlsCustomVerificationCheckRequest cpp_request(&request);
-  EXPECT_EQ(cpp_request.verified_root_cert_subject(), "");
-  grpc::Status sync_status;
-  bool is_sync = verifier->Verify(&cpp_request, nullptr, &sync_status);
-  EXPECT_TRUE(is_sync);
-  EXPECT_TRUE(sync_status.ok())
-      << sync_status.error_code() << " " << sync_status.error_message();
-}
-
 TEST(TlsCertificateVerifierTest, VerifiedRootCertSubjectVerifierSucceeds) {
   grpc_tls_custom_verification_check_request request;
-  std::string expected_subject =
+  constexpr char kExpectedSubject[] =
       "CN=testca,O=Internet Widgits Pty Ltd,ST=Some-State,C=AU";
-  request.peer_info.verified_root_cert_subject = expected_subject.c_str();
+  request.peer_info.verified_root_cert_subject =
+      const_cast<char*>(kExpectedSubject);
   auto verifier =
       ExternalCertificateVerifier::Create<VerifiedRootCertSubjectVerifier>(
-          expected_subject);
+          const_cast<char*>(kExpectedSubject));
```

Comment:
The `const_cast<>` should not be needed here.  It's fine to create an `absl::string_view` from a non-const `char*`.

Reply:
Removed them